crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/service/MaatCalculateContributionServiceTest.java

```diff
@@ -116,28 +116,35 @@ void givenAValidCaseType_whenCalculateContributionIsInvoked_thenShouldNotCalledC
     }
 
     @Test
-    void givenValidContributionAndCompareResultIsLessThanTwo_whenCreateContributionsIsInvoked_thenContributionIsReturn() {
-        when(compareContributionService.compareContribution(
+    void givenValidContributionAndShouldCreateIsTrue_whenCreateContributionsIsInvoked_thenContributionIsReturn() {
+        when(compareContributionService.shouldCreateContribution(
                 any(CalculateContributionDTO.class), any(ContributionResult.class))
-        ).thenReturn(1);
-
+        ).thenReturn(true);
         when(maatCourtDataService.createContribution(any()))
                 .thenReturn(TestModelDataBuilder.getContribution());
+        when(contributionRequestMapper.map(any(CalculateContributionDTO.class), any(ContributionResult.class)))
+                .thenReturn(new CreateContributionRequest());
 
         Contribution result = maatCalculateContributionService.createContributions(new CalculateContributionDTO(),
                 ContributionResult.builder().build()
         );
+
+        verify(maatCourtDataService, times(1))
+                .createContribution(any(CreateContributionRequest.class));
         assertThat(result).isNotNull();
     }
 
     @Test
-    void givenValidContributionAndCompareResultIsGreaterThanTwo_whenCreateContributionsIsInvoked_thenNullIsReturn() {
-        when(compareContributionService.compareContribution(
+    void givenValidContributionAndShouldCreateIsFalse_whenCreateContributionsIsInvoked_thenNullIsReturn() {
```

Comment:
The naming already looks to be a bit bad on these tests (the `thenNullIsReturn`) and the like. If you wanted to fix them up whilst you're renaming them that'd be cool, but up to you.

Reply:
Good point the naming was a bit wonky, I've renamed the tests to hopefully be a bit clearer now.